docs/parse_argparse.py

```diff
@@ -123,57 +117,62 @@ def path_dir_docs_scripts(fpath):
     return (rst_dirpath)
 
 
-def write_run_argparse_output(tmp_fpath, rst_dirpath, fpath):
+def get_argparse_out(argparse_lines):
     """
-    Runs the `--help` command on the agrparse block, copies the output to a rst file
-    with proper identation
+    Write the argparse block to a temporary file and run the `--help` command on it
+    Indents the output using `sed` to be inaccordance with rst syntax
 
     Parameters
     ----------
+    argparse_lines : string
+        argparse block between (and includes) import argparse and 
+        args = parser.parse_args() to be written to a temporary file
+
+    Returns
+    -------
     tmp_fpath : string
         path to temporary file that has the argparse block
         e.g. '</path/to/tmp/file>'
-    rst_dirpath : string
-        path to the directory where script.rst files will be written,
-        e.g. '</path/to/docs/source/scripts/directory>'
+    proc : string
+        indented arparse output after running the `--help` command    
+    """
+    __, tmp_fpath = tempfile.mkstemp()
+    with open(tmp_fpath, 'w') as outfh:
+        outfh.write(argparse_lines)
+    cmd = f"python {tmp_fpath} -h | sed 's/^/   /'"
+    # capture the argparse output
+    proc = subprocess.run(cmd, stdout=subprocess.PIPE,
+                          stderr=subprocess.DEVNULL, shell=True, text=True)
+    return (tmp_fpath, proc)
```

Comment:
I'm not sure why you are taking the extra step when you can just capture the usage information?